### PR TITLE
Update grey color palette and replace color specific variable calls

### PR DESCRIPTION
**Pull request by acostanzo**
[#163805431]

* Update all greys
* Remove white10 and repalce with more descriptive variables
* Repalce grey variable usage with more specific variabls for icons,
backgrounds, text, and borders
* Remove color variants for p, ui, longform since they used the same
value.

### CODE REVIEW

File: src/Checkbox/index.js [hunk 1]
```diff
@@ -42,14 +42,14 @@ const Checkbox = ({
       <Icon
         fontSize="24px"
         name={icon}
-        color={disabled ? 'grey20' : 'grey50'}
+        color={disabled ? 'icon.disabled' : 'icon.default'}
```
Comment (dgalarza) [inline src/Checkbox/index.js]: I wonder if we should actually have a `disabled` prop on `Icon` instead that contains this. Feels painful for everywhere we have a "disabled icon" we'd need to do this similar check.

File: src/Checkbox/index.js [hunk 2]
```diff
@@ -42,14 +42,14 @@ const Checkbox = ({
...
         ml="smaller"
       />
     )}
 
     <Text.span
       fontSize="size2"
-      color={disabled ? 'grey40' : 'grey100'}
+      color={disabled ? 'text.disabled' : 'text.default'}
```
Comment (dgalarza) [inline src/Checkbox/index.js]: Similar question here. Should we just have `disabled` text?

File: src/Checkbox/index.js (hunk 1 above)
Comment (acostanzo) [reply src/Checkbox/index.js]: Good point. I think I like the disabled prop better as well.

Comment (acostanzo) [reply src/Checkbox/index.js]: Based on our chat, we are going to hold off on doing this change. This is specific to the icon next to the label of a checkbox and we were thinking icons as a whole.

**Pull request merged**